src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -130,24 +130,25 @@ bool convert_precision(ov::pass::PassBase& pass,
     };
 
     auto convert_node_output_precision = [&](const std::shared_ptr<ngraph::Node>& node) {
+        bool res = false;
         for (const auto& output : node->outputs()) {
             if (output.get_element_type() == from) {
                 // Handle case with Constants as they can have consumers from other nGraph
                 // Function object
                 auto it = const_to_internal_output.find(node.get());
                 if (it != const_to_internal_output.end()) {
-                    return fuse_type_to_constant(node, to, it->second);
+                    res |= fuse_type_to_constant(node, to, it->second);
+                    continue;
                 }
 
                 // Check that node type exists in map and we can fuse type into node
                 auto t2f_it = type_to_fuse.find(node->get_type_info());
-                if (t2f_it != type_to_fuse.end() && t2f_it->second(node, to, output.get_index())) {
-                    // We need to break if original node was replaced
-                    return true;
+                if (t2f_it != type_to_fuse.end()) {
+                    res |= t2f_it->second(node, to, output.get_index());
```

Comment:
My only concern is about the case, when `replace_node` is called inside `fuse_type_to...` function. 
The loop was previously stopped, after this change the iteration can be continued over the old node outputs.

@itikhono Could you confirm this is the right solution for the https://github.com/openvinotoolkit/openvino/pull/14229#discussion_r1032082670 ?

Reply:
As I can see, if the implementation of `fuse_type_to...` is correct, e.g.
![image](https://user-images.githubusercontent.com/21181928/205666817-1d4d2718-f2c1-4c32-a50f-9477128b39ac.png)
It's not a problem to implement it this way.
We need to check the implementations of other  `fuse_type_to...` functions

Reply:
I've checked the implementations and they all behave the same way.